libs/coin-modules/coin-evm/src/logic/common.ts

```diff
@@ -68,3 +81,36 @@ export function getErc20Data(recipient: string, amount: bigint): Buffer {
   const data = contract.encodeFunctionData("transfer", [recipient, amount]);
   return Buffer.from(data.slice(2), "hex");
 }
+
+export async function prepareUnsignedTxParams(
+  currency: CryptoCurrency,
+  transactionIntent: TransactionIntent<MemoNotSupported>,
+): Promise<TransactionLikeWithPreparedParams> {
+  const { amount, asset, recipient, sender, type } = transactionIntent;
+  const transactionType = getTransactionType(type);
+  const node = getNodeApi(currency);
+
+  const to = isNative(asset) ? recipient : (asset.assetReference as string);
+  const data = isNative(asset) ? Buffer.from([]) : getErc20Data(recipient, amount);
+  const value = isNative(asset) ? amount : 0n;
+
+  const gasLimit = await node.getGasEstimation(
+    { currency, freshAddress: sender },
+    { amount: BigNumber(value.toString()), recipient: to, data },
+  );
+
+  // Single call to fetch feeData
+  const fullFeeData = await node.getFeeData(currency, {
+    type: transactionType,
+    feesStrategy: transactionIntent.feesStrategy,
+  });
+
+  return {
+    type: transactionType,
+    to,
+    data: "0x" + data.toString("hex"),
+    value,
+    gasLimit: gasLimit,
+    feeData: fullFeeData,
+  };
```

Comment:
No big deal, just being more compact
```suggestion
  const feeData = await node.getFeeData(currency, {
    type: transactionType,
    feesStrategy: transactionIntent.feesStrategy,
  });

  return {
    type: transactionType,
    to,
    data: "0x" + data.toString("hex"),
    value,
    gasLimit,
    feeData,
  };
```